### PR TITLE
Pass master info to client relation

### PR DESCRIPTION
**Pull request by ktsakalozos**

### CODE REVIEW

File: bigtop-packages/src/charm/spark/layer-spark/reactive/spark.py [hunk 1]
```diff
@@ -126,10 +127,18 @@ def send_fqdn():
 
 @when('spark.started', 'client.joined')
 def client_present(client):
-    client.set_spark_started()
+    if is_state('leadership.is_leader'):
+        client.set_spark_started()
+        dist = get_dist_config()
+        spark = Spark(dist)
+        master_host = get_fqdn()
+        master_ip = utils.resolve_private_address(master_host)
```
Comment (kwmonroe) [inline bigtop-packages/src/charm/spark/layer-spark/reactive/spark.py]: can we ditch the call to get_fqdn and just use:

```
master_ip = utils.resolve_private_address(hookenv.unit_private_ip())
```

**Comment by pengale** [comment]
Code looks good, and bundle tests pass.

Is there anything that I can do to test this specific change?

**Comment by ktsakalozos** [comment]
You can deploy Spark with Zeppelin and check that it Zeppelin (/etc/zeppelin/conf/zeppelin-env.sh) gets re-configured after a spark master dies. Also, when spark gets in HA mode zeppelin should get reconfigured.... hm... I realised now that I did not test the HA transition because spark HA and the current patch were pending at the same time.

**Comment by kwmonroe** [comment]
LGTM

**Pull request merged**